### ISSUE TITLE
FreeBSD: Partial keyboard support.

### ISSUE THREAD

**Issue by kolbycrouch** [libretro/RetroArch#8591]
## Description

Hello. I've noticed when trying to run RetroArch on FreeBSD or NetBSD, I 
only get a partially working input configuration with my keyboard.

I'm running as root, so no permission problem on device nodes that I'm aware of.
Several keys work, such as: Enter,F,ESC,Backspace and a few others.
Most keys don't work, such as: arrow keys and most other menu hotkeys.

This happens regardless of input driver I use (SDL2, X, libusb ) I've even built against FreeBSD evdev which gives the same results.

The configuration options for the input buttons are correct and match what I have on Linux, 
so it doesn't seem to be a problem there.

I've tried to change the values of input_device_p1 and input_libretro_device_p1 to no avail.

### Expected behavior

Keyboard to be auto-configured as it is in Linux/Windows/OSX/Haiku

### Actual behavior

Only some keys are working on FreeBSD/NetBSD

### Steps to reproduce the bug

Run latest RetroArch

### Version/Commit

- RetroArch: 1.7.6+

### Environment information

- OS: FreeBSD/NetBSD
- Compiler: Clang/GCC

**Comment by orbea** [comment]
I don't think this is specific to FreeBSD/NetBSD and related to how input is handled. There may be some configuration settings that help and maybe duplicate issues as well?

**Comment by niacat** [comment]
I can reproduce this. I cannot find any duplicate issues. For some reason "down" is read as "kp_enter".

Right now I have `menu_unified_controls = "true"` and `input_player1_up/down/left/right` set to WASD, which seems to work. The default configuration is broken.

**Comment by niacat** [comment]
At least part of the problem seems to be that `input/include/xfree86_keycodes.h` assumes that X11 uses evdev keycodes, which is not the case on these systems, as revealed by `xev`.

After correcting the keycodes to those in `${X11BASE}/share/X11/xkb/keycodes/xfree86`, arrow keys miraculously work again.

**Comment by LibretroAdmin** [comment]
PRs welcome
